### PR TITLE
Implement ItemEnhancement4.

### PR DESCRIPTION
**Pull request by Unengine**
Assignment of `ItemEnhancement.ResultModel.gold` was missing, so implemented new `ItemEnhancement` action that covers.

### CODE REVIEW

File: Lib9c/Action/ItemEnhancement4.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+            enhancementEquipment = ItemEnhancement.UpgradeEquipment(enhancementEquipment);
+
+            var requiredBlockIndex = ctx.BlockIndex + RequiredBlockCount;
+            enhancementEquipment.Update(requiredBlockIndex);
+            sw.Stop();
+            Log.Debug("ItemEnhancement Upgrade Equipment: {Elapsed}", sw.Elapsed);
+            sw.Restart();
+
+            result.gold = requiredNCG;
```
Comment (Unengine) [inline Lib9c/Action/ItemEnhancement4.cs:201]:
```suggestion
            result.gold = 0;
```

This line is changed.

**Review by boscohyun — APPROVED** [review]

**Comment by longfin** [comment]
It would be better we have unit tests about it.

**Comment by riemannulus** [comment]
Base branch seems outdated 😢

File: .Lib9c.Tests/Action/ItemEnhancement4Test.cs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+            var grade = resultEquipment.Grade;
+            var costRow = _tableSheets.EnhancementCostSheet
+                .OrderedList
+                .FirstOrDefault(x => x.Grade == grade && x.Level == resultEquipment.level);
+            var stateDict = (Dictionary)nextState.GetState(slotAddress);
+            var slot = new CombinationSlotState(stateDict);
+            var slotResult = (ItemEnhancement.ResultModel)slot.Result;
+
+            Assert.Equal(costRow.Cost, slotResult.gold);
```
Comment (ipdae) [inline .Lib9c.Tests/Action/ItemEnhancement4Test.cs:147]:
```suggestion
            Assert.Equal(0, slotResult.gold);
```

Comment (Unengine) [reply .Lib9c.Tests/Action/ItemEnhancement4Test.cs:147]: 액션에서 `slotResult.gold`를 잘 넣어주는지 확인하기 위한 의도였는데 문제가 될까요?

**Review by ipdae — APPROVED** [review]

**Pull request merged**